extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/DefaultConfigExtractorUtil.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.opentelemetry.deployment;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.quarkus.deployment.configuration.definition.ClassDefinition;
+import io.quarkus.deployment.configuration.definition.RootDefinition;
+
+public class DefaultConfigExtractorUtil {
+
+    public static Map<String, String> extractDefaultKeyValues(final RootDefinition rootDefinition) {
+        Map<String, String> pathsAndValues = new HashMap<>();
+        if (rootDefinition == null) {
+            return pathsAndValues;
+        }
+        Iterable<ClassDefinition.ClassMember> members = rootDefinition.getMembers();
+        for (ClassDefinition.ClassMember member : members) {
+            extractPaths("" + rootDefinition.getPrefix() + "." + rootDefinition.getRootName(), pathsAndValues, member);
+        }
+        return pathsAndValues;
+    }
+
+    private static void extractPaths(String path, Map<String, String> pathsAndValues, ClassDefinition.ClassMember member) {
+        String propertyName = path + "." + member.getPropertyName();
+
+        if (member instanceof ClassDefinition.ItemMember) {
+            // leaf
+            String defaultValue = ((ClassDefinition.ItemMember) member).getDefaultValue();
+            if (defaultValue == null) {
+                return;
+            }
+            pathsAndValues.put(propertyName, defaultValue);
+            return;
```

Comment:
We don't actually need this return right? since we are in a if-else flow =]

Reply:
Correct, but as I was writing it I didn't knew if I was going to write something else after the IFs.
See next comment.